messaging/src/main/java/org/axonframework/eventhandling/WrappedToken.java

```diff
@@ -16,19 +16,51 @@
 
 package org.axonframework.eventhandling;
 
+import java.util.Optional;
+
 /**
  * Interface making a token that wraps another token. As certain implementations may depend on specific token types,
  * Tokens that wrap another must provide a means to retrieve the original token.
  *
  * @author Allard Buijze
  * @since 3.2
  */
-public interface WrappedToken {
+public interface WrappedToken extends TrackingToken {
+
+    static TrackingToken unwrapLowerBound(TrackingToken token) {
+        return token instanceof WrappedToken ? ((WrappedToken) token).lowerBound() : token;
+    }
+
+    static TrackingToken unwrapUpperBound(TrackingToken token) {
+        return token instanceof WrappedToken ? ((WrappedToken) token).upperBound() : token;
+    }
+
+    static <R extends TrackingToken> Optional<R> unwrap(TrackingToken token, Class<R> tokenType) {
+        if (token instanceof WrappedToken) {
+            return ((WrappedToken) token).unwrap(tokenType);
+        } else if (tokenType.isInstance(token)) {
+            return Optional.of(tokenType.cast(token));
+        } else {
+            return Optional.empty();
+        }
+    }
+
+    /**
+     * Advance this token to the given {@code newToken}.
+     *
+     * @param newToken The token representing the position to advance to
+     * @return a token representing the new position
+     */
+    TrackingToken advancedTo(TrackingToken newToken);
 
     /**
      * Returns the token representing the current position in the stream.
      *
      * @return the token representing the current position in the stream
      */
-    TrackingToken unwrap();
```

Comment:
Breaking API change?

Reply:
Yes, breaking. But I don't expect anyone to have built a custom implementation of a WrappedToken. And if they have (which we could consider an extreme customization), at least there is a compile error.